### PR TITLE
Update to support S32K JLink scripting.

### PR DESCRIPTION
**Pull request by redxeth**
Minor updates to permit changing the script interfacee method for JLink scripts.  Added support for mem32, q (quit) and halt commands.

### CODE REVIEW

File: lib/origen_debuggers/j_link.rb [hunk 1]
```diff
@@ -245,6 +245,11 @@ def read32(data, options = {})
       alias_method :read_longword, :read32
       alias_method :read_32, :read32
 
+      # Read 32-bit chunks of data using given byte address
```
Comment (ginty) [inline lib/origen_debuggers/j_link.rb]: Don't really like adding a new method for something like this, better to make the existing read32 method accept a single value or an array of values for the data arg

File: lib/origen_debuggers/j_link.rb [hunk 2]
```diff
@@ -341,11 +346,43 @@ def read_ir(reg_or_val, options = {})
...
-          number_of_bytes: 1
+          number: 1,    # number of items to read
+          type:   :byte           # type of data to be read
         }.merge(options)
-        dw "mem 0x#{address.to_s(16).upcase}, #{options[:number_of_bytes]}"
+
+        # for backward compatibility
+        options[:number] = options[:number_of_bytes] if options[:number_of_bytes]
+
+        if options[:type] == :_32
```
Comment (ginty) [inline lib/origen_debuggers/j_link.rb]: This should be called :size with a numeric arg instead of type (to align to generally used conventions)

**Review by ginty — CHANGES_REQUESTED** [review]

File: lib/origen_debuggers/j_link.rb (hunk 2 above)
Comment (redxeth) [reply lib/origen_debuggers/j_link.rb]: There are discrete memXX commands so can't really have user enter anything they want.   But could go numeric and just give error if invalid value.

Comment (ginty) [reply lib/origen_debuggers/j_link.rb]: Yes that's fine and would be the conventional way to do it

File: lib/origen_debuggers/j_link.rb (hunk 1 above)
Comment (redxeth) [reply lib/origen_debuggers/j_link.rb]: Was trying to get something in there quick -- but can try that.

**Review by ginty — APPROVED** [review]

**Pull request merged**